File: src/main/js/werkboek/week1/les1/1_5.js
class Student {
    #name;
    #nummer;

    constructor(nm, nr = 0) {
        this.#name = nm;
        this.#nummer = nr;
    }

    get naam() {
        return this.#name;
    }

    get studentnr() {
        return this.#nummer;
    }

    set studentnr(nr) {
        this.#nummer = nr;
    }

    toString() {
        return "Deze student heet " + this.#name + " en heeft nummer " + this.#nummer;
    }
}

let st1 = new Student("Sylvester");

console.log("naam: " + st1.naam);
console.log("nummer: " + st1.studentnr);
st1.studentnr = 12345678;
console.log("toString: " + st1.toString());
console.log();

let st2 = new Student("Karel", 98765432);
console.log("naam: " + st2.naam);
console.log("nummer: " + st2.studentnr);
console.log("toString: " + st2.toString());

File: src/main/python/werkboek/week1/les1/1_5.py
class Student:
    def __init__(self, name, nr=0):
        self._name = name  # private wordt ongeveer opgelost met een underscore in python, als je dat echt wilt
        self.studentnr = nr

    def __str__(self):
        return "Deze student heet " + self._name + " en heeft nr " + str(self.studentnr)

    @property  # Dit is hoe je hetzelfde gedrag als in Java krijgt
    def name(self):
        return self._name


student1 = Student("Sylvester")
student2 = Student("Karel", 98765432)

print("naam: " + student1.name)
print("nummer: " + str(student1.studentnr))
student1.studentnr = 12345678 # Python heeft dus geen 'setXYZ' methodes zoals Java
print("__to_s__: " + str(student1))
print("")

print("naam: " + student2.name)
print("nummer: " + str(student2.studentnr))
print("__to_s__: " + str(student2))
